README.md

```diff
@@ -1 +1,961 @@
+4. [Build the `helloworld` package](#build-the-helloworld-package)
+
+### The "OmniOS Extra Build System"
+
+1. [Package naming conventions](#package-naming-conventions)
+2. [Structure of an OmniOS package](#structure-of-an-omnios-package)
+3. [The `local.mog` and `lib/global-transforms.mog` files](#the-local-mog-and-lib-global-transforms-mog-files)
+4. [Build and run dependencies](#build-and-run-dependencies)
+5. [Configure directives](#configure-directives)
+6. [Make directives](#make-directives)
+7. [Linker directives](#linker-directives)
+8. [Using your own functions in `build.sh`](#using-your-own-functions-in-build-sh)
+9. [Creating patches](#creating-patches)
+10. [SMF manifests](#smf-manifests)
+11. [Providing messages at installation](#providing-messages-at-installation)
+12. [Making the source available to the build environment](#making-the-source-available-to-the-build-environment)
+13. [Managing licences](#managing-licences)
+14. [Adding package information to the build system](#adding-package-information-to-the-build-system)
+15. [The build system's default "IPS Repository"](#the-build-system-s-default-ips-repository)
+
+### Tips for building specific types of packages
+
+1. [Building packages that share common elements](#building-packages-that-share-common-elements)
+2. [Tips for building libraries](#tips-for-building-libraries)
+3. [Tips for Go packages](#tips-for-go-packages)
+4. [Tips for Perl packages](#tips-for-perl-packages)
+5. [Tips for Python packages](#tips-for-python-packages)
+6. [Tips for Rust packages](#tips-for-rust-packages)
+
+### Submit your package to the "OmniOS Extra IPS Repository"
+
+1. [Create a "Pull Request" for the "OmniOS Extra IPS Repository"](#create-a-pull-request-for-the-omnios-extra-ips-repository)
+
+### Getting Help
+
+1. [OmniOS community support channels](#omnios-community-support-channels)
+2. [Recommended reading](#recommended-reading)
+
+# Prepare to build packages for OmniOS
+The objective of this section is to prepare a build environment and introduce the `build.sh` and `local.mog` files. After this, an experimental build can be run with an example package.
+
+1. [Requirements](#requirements)
+2. [Create the build environment](#create-the-build-environment)
+3. [Create a build directory and it's build files](#create-a-build-directory-and-its-build-files)
+4. [Build the `helloworld` package](#build-the-helloworld-package)
+
+## Requirements
+
+To create packages for the "OmniOS Extra IPS Repository", the following is necessary:
+
+* Virtual or  physical system running the latest OmniOS release, preferably the  latest ["Bloody Release"](https://omniosce.org/about/stablevsbloody).
+* The system should have a minimum of 2GB of RAM.
+
+* "OmniOS [Build Tools](https://github.com/omniosorg/omnios-build/blob/master/build/meta/omnios-build-tools.p5m)": this can be installed with the following command:
+
+```none
+# pkg install ooce/omnios-build
```

Comment:
It's just `ooce/omnios-build-tools` that's required, you removed the wrong one, and lost the `-tools` on the install command.

Reply:
I'm confused: That link points to omnios-build-tools. That is the only required package, no? 
The command I get, I can add the trailing `-tools`  once the above is confirmed.

Reply:
Sorry, I made a mistake too.

The only package you need to install for working on extra, is `ooce/extra-build-tools`
https://github.com/omniosorg/omnios-extra/blob/master/build/meta/extra-build-tools.p5m

This will automatically pull in everything that's required.

Reply:
that is sorted.